File: apps/chat-android/messaging-websocket/src/main/java/com/amazonaws/services/chime/sdk/messaging/internal/WebSocketAdapter.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package com.amazonaws.services.chime.sdk.messaging.internal

interface WebSocketAdapter {
    fun create(url: String, observer: WebSocketAdapterObserver)

    fun close()
}

File: apps/chat-ios/AmazonChimeSDKMessagingDemo/messaging-websocket/internal/WebSocketAdapter.swift
//
//  WebSocketAdapter.swift
//  AmazonChimeSDKMessagingDemo
//
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: MIT-0
//

public protocol WebSocketAdapter {
    func connect(url: String, observer: WebSocketAdapterObserver)

    func close()
}
